src/core/composer/qgscomposermap.cpp

```diff
@@ -2055,7 +2032,7 @@ QString QgsComposerMap::displayName() const
 void QgsComposerMap::requestedExtent( QgsRectangle& extent ) const
 {
   QgsRectangle newExtent = *currentMapExtent();
-  if ( mEvaluatedMapRotation == 0 )
+  if ( 1 || mEvaluatedMapRotation == 0 )
```

Comment:
@strk I don't think this change should be included. It breaks API and affects QgsComposerLegend::updateFilterByMap() potentially causing features to be missed in a legend filtered from a rotated map (although the current behaviour isn't ideal as it potentially includes features which aren't visible, but that's safer then the alternative). Instead, the calls to requestedExtent in cache and paint should be replaced by *currentMapExtent() instead.